Python/RanjanPaudel/task1/store_result.py

```diff
@@ -0,0 +1,79 @@
+#!/usr/bin/env python
+
+import csv
+import argparse
+import datetime
+import pathlib
+
+USAGE_EXAMPLE = "Usage example:\n python3 store_result.py --name John Doe --dob 01-11-2001 --subject Nepali --total 100 --score 65.25 --store store.csv"
+NAME_HELP = "Name of student: 'FIRST MIDDLE(optional) LAST'"
+DOB_HELP = "Student's Date of Birth in DD-MM-YYYY format."
+SUBJECT_HELP = "Value should be from the set."
+TOTAL_HELP = "Total/full marks for the subject."
+SCORE_HELP = "Score of the selected subject. Value should be a floating point number and can not be greater than full/total marks of the subject or less than 0."
+STORE_HELP = "File name to store the result. e.g. store.csv"
+
+FIELD_NAMES = ['name', 'dob', 'subject', 'total', 'score', 'percentage']
+
+
+def validate_date(date_string):
+    try:
+        datetime.datetime.strptime(date_string, '%d-%m-%Y')
+        return date_string
+    except (ValueError):
+        raise SystemExit('Invalid date format. Should be DD-MM-YYYY.')
+
+
+parser = argparse.ArgumentParser(description=USAGE_EXAMPLE,
+                                 formatter_class=argparse.RawDescriptionHelpFormatter)
+parser.add_argument("--name",
+                    help=NAME_HELP,
+                    required=True)
+parser.add_argument("--dob", type=validate_date,
+                    help=DOB_HELP,
+                    required=True)
+parser.add_argument("--subject",
+                    choices=['Nepali', 'English', 'Math', 'Science', 'Social'],
+                    help=SUBJECT_HELP,
+                    required=True)
+parser.add_argument("--total", type=int,
+                    help=TOTAL_HELP,
+                    required=True)
+parser.add_argument("--score", type=float,
+                    help=SCORE_HELP,
+                    required=True)
+parser.add_argument("--store",
+                    help=STORE_HELP,
+                    required=True)
+
+
+if __name__ == "__main__":
+    opts = parser.parse_args()
+    result_dict = opts.__dict__
+    file_name = result_dict.pop('store')
+
+    if result_dict['score'] > result_dict['total'] or result_dict['score'] < 0 or result_dict['score'] == -0.0:
```

Comment:
this can be a validation function and can be provided while adding argument in argparse